File: VS2012/ASP.NET.MVC.NUnit.ProjectTemplate.v4.vb.WebApi/Controllers/ValuesControllerTest.vb
Imports System
Imports System.Collections.Generic
Imports System.Net.Http
Imports System.Text
Imports System.Web.Http
Imports NUnit.Framework
Imports $mvcprojectnamespace$

Imports NIs = NUnit.Framework.Is

<TestFixture()> Public Class ValuesControllerTest
    <Test()> Public Sub GetValues()
        'Arrange
        Dim controller As New ValuesController()

        'Act
        Dim result As IEnumerable(Of String) = controller.GetValues()

        'Assert
        Assert.That(result, NIs.Not.Null)
        Assert.That(2, NIs.EqualTo(result.Count()))
        Assert.That("value1", NIs.EqualTo(result.ElementAt(0)))
        Assert.That("value2", NIs.EqualTo(result.ElementAt(1)))
    End Sub

    <Test()> Public Sub GetValueById()
        'Arrange
        Dim controller As New ValuesController()

        'Act
        Dim result As String = controller.GetValue(5)

        'Assert
        Assert.That("value", NIs.EqualTo(result))
    End Sub

    <Test()> Public Sub PostValue()
        'Arrange
        Dim controller As New ValuesController()

        'Act
        controller.PostValue("value")

        'Assert
    End Sub

    <Test()> Public Sub PutValue()
        'Arrange
        Dim controller As New ValuesController()

        'Act
        controller.PutValue(5, "value")

        'Assert
    End Sub

    <Test()> Public Sub DeleteValue()
        'Arrange
        Dim controller As New ValuesController()

        'Act
        controller.DeleteValue(5)

        'Assert
    End Sub
End Class

File: VS2012/ASP.NET.MVC.NUnit.ProjectTemplate.v4.cs.WebApi/Controllers/ValuesControllerTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Web.Http;
using NUnit.Framework;
using $mvcprojectnamespace$;
using $mvcprojectnamespace$.Controllers;

namespace $safeprojectname$.Controllers
{
    [TestFixture]
    public class ValuesControllerTest
    {
        [Test]
        public void Get()
        {
            // Arrange
            ValuesController controller = new ValuesController();

            // Act
            IEnumerable<string> result = controller.Get();

            // Assert
            Assert.That(result, Is.Not.Null);
            Assert.That(2, Is.EqualTo(result.Count()));
            Assert.That("value1", Is.EqualTo(result.ElementAt(0)));
            Assert.That("value2", Is.EqualTo(result.ElementAt(1)));
        }

        [Test]
        public void GetById()
        {
            // Arrange
            ValuesController controller = new ValuesController();

            // Act
            string result = controller.Get(5);

            // Assert
            Assert.That("value", Is.EqualTo(result));
        }

        [Test]
        public void Post()
        {
            // Arrange
            ValuesController controller = new ValuesController();

            // Act
            controller.Post("value");

            // Assert
        }

        [Test]
        public void Put()
        {
            // Arrange
            ValuesController controller = new ValuesController();

            // Act
            controller.Put(5, "value");

            // Assert
        }

        [Test]
        public void Delete()
        {
            // Arrange
            ValuesController controller = new ValuesController();

            // Act
            controller.Delete(5);

            // Assert
        }
    }
}
